src/SemanticMediaWiki/DataAnnotator.php

```diff
@@ -69,15 +69,29 @@ public function addAnnotations( SDTopic $topic, SemanticData $semanticData ): vo
 	 * @param SDTopic $topic
 	 */
 	private function addRepliesAnnotations( array $replies, SemanticData $semanticData, SDTopic $topic ): void {
-		foreach ( $replies as $reply ) {
+		foreach ( $replies as $index => $reply ) {
 			if ( !$reply->isEveryoneAllowed() ) {
 				// Do not annotate the reply if it is not viewable by everyone, since this WILL lead to
 				// information leakage
 				continue;
 			}
-
 			$id = sprintf( 'flow-post-%s', $reply->getPostId() );
 
+			// check if the reply should be saved for searchresults
+			$shouldSaveReply = true;
+			$this->annotatorStore->hookRunner->onSemanticStructuredDiscussionsShouldSaveReply(
```

Comment:
```suggestion
			$this->hookRunner->onSemanticStructuredDiscussionsShouldSaveReply(
```

Reply:
Instead of relying on the `HookRunner` in the annotator store, I would suggest using the service wiring to inject the hook runner directly:

1. Add the hook runner to the signature of the constructor:
   ```php
   public function __construct( AnnotatorStore $annotatorStore, HookRunner $hookRunner )
   ```
2. Add the dependency to the service wiring file `SemanticStructuredDiscussions.ServiceWiring.php`.